public/app/features/plugins/admin/hooks/usePluginDetailsTabs.tsx

```diff
@@ -16,13 +16,29 @@ type ReturnType = {
   activePageId: PluginTabIds | string;
 };
 
-export const usePluginDetailsTabs = (plugin?: CatalogPlugin, pageId?: PluginTabIds): ReturnType => {
+export const usePluginDetailsTabs = (
+  plugin?: CatalogPlugin,
+  pageId?: PluginTabIds,
+  isNarrowScreen?: boolean
+): ReturnType => {
   const { loading, error, value: pluginConfig } = usePluginConfig(plugin);
   const { pathname } = useLocation();
   const defaultTab = useDefaultPage(plugin, pluginConfig);
   const isPublished = Boolean(plugin?.isPublished);
 
-  const currentPageId = pageId || defaultTab;
+  function getCurrentPageId(
+    pageId: PluginTabIds | undefined,
+    isNarrowScreen: boolean | undefined,
+    defaultTab: string
+  ): PluginTabIds | string {
+    if (!isNarrowScreen && pageId === PluginTabIds.PLUGINDETAILS) {
+      return defaultTab;
+    }
+    return pageId || defaultTab;
+  }
```

Comment:
Nit: This can be moved outside of the hook so it doesn't get declared on every render.